### ISSUE TITLE
Source code was not found for app:///main.jsbundle

### ISSUE THREAD

**Issue by dengwanc** [getsentry/react-native-sentry#338]
OS:
- [ ]  Windows
- [x]  MacOS
- [ ]  Linux

_Platform:_
- [x]  iOS
- [ ]  Android

Output of `node -v && npm -v && npm ls --prod --depth=0`

```
v6.12.1
3.10.10
MyReactNativeApp@0.0.1 /Users/dengwanc/Projects/maimai_react_native
├── immutable@3.8.1
├── moment-mini@2.18.1
├── prop-types@15.5.10
├── query-string@5.0.0
├── react@16.0.0-alpha.12
├── react-native@0.47.1
├── react-native-iphone-x-helper@1.0.1
├── react-native-linear-gradient@2.2.0
├── react-native-popup-menu@0.8.0
├── react-native-sentry@0.32.1
├── react-redux@5.0.5
├── redux@3.7.2
├── redux-actions@2.2.1
├── redux-batched-actions@0.1.6
└── underscore@1.8.3
```

Config:

```
Sentry.config("http://2080b86958ae4180ae1203902f735dc0:fa100f295ecd4d9389b5220d2cd9d76a@sentry.mm.taou.com//18", {
    release: '2da95dfb052f477380608d59d32b4ab9',
}).install();
```

---
I have following issue:

react native sentry source map JUST not working !!!

my upload script 

```
sentry-cli --auth-token 27d37b4efd73439d820f02561e4507ca7d917ee771f84ed1949148137c3fb9a3 \
--url http://sentry.mm.taou.com \
releases -o sentry -p react-native files \
2da95dfb052f477380608d59d32b4ab9 upload-sourcemaps \
--rewrite --validate main.jsbundle.map
```

I GOT THIS IN Senry UI ，I am SO confused NOW

```
Source code was not found for app:///main.jsbundle 
{
  "url": "app:///main.jsbundle"
}
```

**Comment by HazAT** [comment]
I guess you have to add `--url-prefix ~/` as an additional parameter.
Can you share a screenshot of your release artifacts in Sentry?

**Comment by dengwanc** [comment]
This is My release list

```
app:/main.jsbundle.map
None
3.8 MB
file:/main.jsbundle
None
1.0 MB
file:/main.jsbundle.map
None
3.8 MB
https://dengwanc.github.io/main.jsbundle
None
1.0 MB
https://dengwanc.github.io/main.jsbundle.map
None
3.8 MB
~/main.jsbundle
None
1.0 MB
~/main.jsbundle.map
None
3.8 MB
```

**Comment by dengwanc** [comment]
and this a demo

https://sentry.io/taou/react-native/issues/439897225/

why sourcemap look strangely like that ?

**Comment by HazAT** [comment]
@dengwanc 
A few things I've noticed.
First, it says reported with `raven-js` as SDK instead of `react-native-sentry` which indicates that something with your initial setup went wrong.
Did you call `react-native link react-native-sentry`?
Also, how are you building your app, in debug or release? Because the stacktrace looks more than strange, almost if you used not the facebook bundler.

**Comment by dengwanc** [comment]
react-native how to point ios and android dir. like

`react-native link react-native-sentry --native-dir my_ios_dir my_android_dir`

**Comment by yaronlevi** [comment]
I followed the instructions carefully (including the code push part):
https://docs.sentry.io/clients/react-native/

And I still can't get the line numbers and source code to be shown at the dashboard. I get this error:

**Source code was not found for app:///index.bundle**

It looks like Sentry is looking for index.bundle, while the artifacts in the release (which was created by Sentry Cli with code push command) looks like this:

**~/main.jsbundle
~/main.jsbundle.map**

**Comment by HazAT** [comment]
@yaronlevi Can you link an event here?
The version and dist of the release in Sentry must match the version and dist on the event itself.
But in general this looks correct, would need to link to give a better answer.

**Comment by anandprabhu** [comment]
@HazAT I'm facing a similar error when creating Android release builds. The `~/index.android.bundle` and the `~/index.android.bundle.map` are present in the Artifacts tab of Releases. But still for the javascript errors it shows `Source code was not found for app://index.android.bundle`. I am using the online hosted version of Sentry.

**Comment by HazAT** [comment]
If your android event it reported with `raven-js`
This is not correct, this indicates something went wrong with the native Android SDK setup.

Here is probably the fix for your problem:
https://github.com/getsentry/react-native-sentry/issues/167
https://github.com/getsentry/react-native-sentry/issues/213

What version of Android Studio and build tools are you running?
I still need a way to reproduce this issue.

Make sure the event is reported with `sentry-react-native`
you can find this on the bottom of the event in the SDK section.

**Comment by anandprabhu** [comment]
@HazAT I have configure the `react-native-sentry` library for both the iOS and Android builds. I am not sure why the SDK is mentioned as `raven-js` in the dashboard. Below are the details of the libs used,

```
"react": "^16.2.0",
"react-native": "^0.52.2",
"react-native-sentry": "^0.33.0"
```
I am using the command line to build the app and not using the Android Studio. Although the version i have currently installed is Android Studio 3.0 and the Build Tools is `26.0.0`. Do let me know if you need more information which i could provide over e-mail directly.

**Comment by HazAT** [comment]
@anandprabhu Can you look at the 2 issues I posted.
It looks like that `compile project(':react-native-sentry')` is missing in `build.gradle`.
The Android SDK is definitely not running and that's why it states `raven-js`.

**Comment by anandprabhu** [comment]
@HazAT I have gone through both the issues but i do not find any issues with the `build.gradle` for my app. It ssems to contain the correct entry for the `compile project(':react-native-sentry')` and creating build also works fine without any issues. I have tried to unlink the library, remove the `node_modules` folder and then do everything from the start. Still facing the same issues.

**Comment by HazAT** [comment]
@anandprabhu 
Can you do the following:
Create a clean react-native project and see if you still having this issue.
```
react-native init AwesomeProject
cd AwesomeProject
yarn add react-native-sentry
react-native link react-native-sentry
react-native run-android
```
I just did it and it worked.
Check the log output if you can find something like
```
...
io.sentry.android.AndroidSentryClientFactory: Sentry init with ctx='com.awesomeproject.MainApplication@95354ef' and dsn='Dsn{uri=https://sentry.io/}'
...
react-native-sentry: startWithDsnString 'YOUR DSN'
```

**Comment by anandprabhu** [comment]
@HazAT I tried the same steps but do not find the lines that you have mentioned at the end. Will it be printout out in the command-line during the `react-native run-android`?

**Comment by HazAT** [comment]
If it is running on a device you have to start `logcat` to see the output.
Alternatively, you can start the project in Android Studio.

**Comment by anandprabhu** [comment]
@HazAT Could if be that our project is mainly created in Typescript and then transpiled into JavaScript code. But we are able to see that the bundle file is being created from the JavaScript source only and uploaded correctly in Sentry. But really not sure why the dashboard mentions `raven-js` in the SDK.

I am able to see the above mentioned entries in the new project created using the steps you had mentioned. But the same is not available in our application using the same libraries. Could it be because of the increment to the `compileSdkVersion 26` and `buildToolsVersion '26.0.0'` ?

**Comment by HazAT** [comment]
It could be the `compileSdkVersion 26` and `buildToolsVersion '26.0.0'` I am not really sure.
The only thing I can say is that `sentry-android` is not running in your project, I have to take a look at your project to debug it otherwise you can only try to make Sentry run in your project.
The issue you are having cannot be Typescript, when your event is sent with `sentry-react-native` the stack trace will be fixed since we `sentry-android` makes sure that we add the necessary release information to the event.

**Comment by anandprabhu** [comment]
@HazAT Ok fine let me try downgrading the version of the Android SDK and Build Tools and check if it works and update you.

**Comment by anandprabhu** [comment]
@HazAT Downgrading the `compileSdkVersion 26` and `buildToolsVersion '26.0.0'` breaks other libraries like `lottie-react-native` and `react-native-vector-icons`. I am unable to trigger a build from the command like or Android Studio. However if i increase the version directly in the Sentry SDK using Android Studio to 26 i am able to create the build but no information is present int he logcat.

**Comment by HazAT** [comment]
@anandprabhu Sorry, with a reproducible case I can't do anything.
It must be something with your setup, all I can say is that the native android package isn't running in your project, why?
For that, I would need to see your project to help you debug it.

**Comment by HazAT** [comment]
@anandprabhu Can you check if your MainApplication.java has these two lines:
```java
...
import io.sentry.RNSentryPackage; // < --------------------
...
public class MainApplication extends Application implements ReactApplication {
    ...
    @Override
    protected List<ReactPackage> getPackages() {
      return Arrays.<ReactPackage>asList(
          new MainReactPackage(),
          new RNSentryPackage(MainApplication.this) // <------------------------------
      );
    }
    ...

```

**Comment by anandprabhu** [comment]
@HazAT The `MainApplication.java` was missing the lines you had mentioned. Once i added the same it works fine now and able to see the mapping. But it seems to be showing it in a wrong line-number though.

**Comment by HazAT** [comment]
@anandprabhu Your uploaded source map should do this correctly.
Keep in mind, old events will not get symbolicated only new ones.
Release and Dist have to match your artifacts.
Also, there is a bug in react-native itself facebook is aware of that line number may be wrong for android builds ref:
https://github.com/getsentry/react-native-sentry/issues/258
https://github.com/bugsnag/bugsnag-react-native/issues/142
https://github.com/facebook/react-native/issues/6946

I am closing this issue now since it's already hard to follow.
If anyone still having an issue with this, please consider creating a new issue with a reproducible case.
It seems like the issue is the linking process of react-native where it's not able to patch MainApplication.java.

**Comment by vivek-walecha-657** [comment]
@dengwanc i am also facing this issue brother not able to see the line number and only seeing this :
Source code was not found for app:///index.android.bundle Collapse
{
  "url": "app:///index.android.bundle"
}

dont know what are the issues under lying.
